File: students/anne_homann/mtalab.rb

# nline: ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th']
# is the new syntax used in Ruby
# old one is being used below

subway = {
  "N_line" => ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'], 
  "L_line" => ['8th', '6th', 'Union Square', '3rd', '1st'],
  "Six_line" => ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
}


puts "Which line are you taking ? #{subway.keys}"
start_line = gets.chomp
puts "From which stop? #{subway[start_line]}"
start_station = gets.chomp
puts "Which line will you be getting off from? #{subway.keys}"
end_line = gets.chomp
puts "And which stop will you be hopping off? #{subway[end_line]}"
end_station = gets.chomp





puts "There will be a total of #{[]} stops before you reach #{subway[end_station]}"

File: students/anne_homann/mta_lab_js/js/mtalab.js
// Subway lines and stops
var nline = ["Times Square", "34th", "28th", "23rd", "Union Square", "8th st"];
var lline = ["8th Ave", "6th", "Union Square", "3rd", "1st"];
var sixline = ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"];

var boardStop = prompt("At which stop are you boarding?");
var boardLine = prompt("Which line is that on?");

var offStop = prompt("At which stop are you de-boarding?");
var offLine = prompt("Which line is that on?");

// Distance from boardStop to Union Square
switch(boardLine) {
  case 'N':
  var toUS = Math.abs(nline.indexOf(boardStop)-nline.indexOf("Union Square"));
  break;
  case 'L':
  var toUS = Math.abs(lline.indexOf(boardStop)-lline.indexOf("Union Square"));
  break;
  case '6':
  var toUS = Math.abs(sixline.indexOf(boardStop)-sixline.indexOf("Union Square"));
  break;
}

// Distance from Union Square to unloadStop
switch(offLine) {
  case 'N':
  var fromUS = Math.abs(nline.indexOf(offStop)-nline.indexOf("Union Square"));
  break;
  case 'L':
  var fromUS = Math.abs(lline.indexOf(offStop)-lline.indexOf("Union Square"));
  break;
  case '6':
  var fromUS = Math.abs(sixline.indexOf(offStop)-sixline.indexOf("Union Square"));
  break;
}

var distance = Math.abs(toUS) + Math.abs(fromUS);
console.log("You will travel "+ distance + " stops");
